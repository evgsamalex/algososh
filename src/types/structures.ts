import {ElementStates} from "./element-states";

export interface IElement<T> {
  value: T,
  state: ElementStates;
  index: number;
  nextState: () => void;
}

export class Element<T> implements IElement<T> {
  constructor(index: number, value: T, state: ElementStates = ElementStates.Default) {
    this.index = index;
    this.value = value;
    this.state = state;
  }

  public value: T;
  public state: ElementStates

  public index: number;

  public nextState() {
    switch (this.state) {
      case ElementStates.Default:
        this.state = ElementStates.Changing;
        break;
      case ElementStates.Changing:
        this.state = ElementStates.Modified;
        break;
    }
  }
}


