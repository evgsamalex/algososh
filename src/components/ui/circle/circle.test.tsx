import renderer from "react-test-renderer";
import {Circle} from "./circle";
import {ElementStates} from "../../../types/element-states";

it('Кружок без буквы', () => {
  const tree = renderer
    .create(<Circle/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Кружок с буквами', () => {
  const tree = renderer
    .create(<Circle letter={'123'}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Кружок с head', () => {
  const tree = renderer
    .create(<Circle letter={'H'} head={'h'}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Кружок с react-элементом в head', () => {
  const tree = renderer
    .create(<Circle letter={'H'} head={<Circle isSmall letter={'h'}/>}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Кружок с tail', () => {
  const tree = renderer
    .create(<Circle letter={'T'} tail={'t'}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Кружок с react-элементом в tail', () => {
  const tree = renderer
    .create(<Circle letter={'T'} tailType={"element"} tail={<Circle letter={'t'}/>}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Кружок c index', () => {
  const tree = renderer
    .create(<Circle index={1}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Кружок с пропcом isSmall ===  true', () => {
  const tree = renderer
    .create(<Circle isSmall/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Кружок default', () => {
  const tree = renderer
    .create(<Circle state={ElementStates.Default}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Кружок changing', () => {
  const tree = renderer
    .create(<Circle state={ElementStates.Changing}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Кружок modified', () => {
  const tree = renderer
    .create(<Circle state={ElementStates.Modified}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
