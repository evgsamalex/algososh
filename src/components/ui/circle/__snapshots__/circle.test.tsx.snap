// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Кружок c index 1`] = `
<div
  className="content "
  data-cy="circle"
  data-test-state="default"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head
          element"
    data-cy="head"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4
          absolute
          tail60
          element
        "
    data-cy="tail"
  />
</div>
`;

exports[`Кружок changing 1`] = `
<div
  className="content "
  data-cy="circle"
  data-test-state="changing"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head
          element"
    data-cy="head"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4
          absolute
          tail30
          element
        "
    data-cy="tail"
  />
</div>
`;

exports[`Кружок default 1`] = `
<div
  className="content "
  data-cy="circle"
  data-test-state="default"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head
          element"
    data-cy="head"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4
          absolute
          tail30
          element
        "
    data-cy="tail"
  />
</div>
`;

exports[`Кружок modified 1`] = `
<div
  className="content "
  data-cy="circle"
  data-test-state="modified"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head
          element"
    data-cy="head"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4
          absolute
          tail30
          element
        "
    data-cy="tail"
  />
</div>
`;

exports[`Кружок без буквы 1`] = `
<div
  className="content "
  data-cy="circle"
  data-test-state="default"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head
          element"
    data-cy="head"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4
          absolute
          tail30
          element
        "
    data-cy="tail"
  />
</div>
`;

exports[`Кружок с head 1`] = `
<div
  className="content "
  data-cy="circle"
  data-test-state="default"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head
          string"
    data-cy="head"
  >
    h
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    >
      H
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4
          absolute
          tail30
          element
        "
    data-cy="tail"
  />
</div>
`;

exports[`Кружок с react-элементом в head 1`] = `
<div
  className="content "
  data-cy="circle"
  data-test-state="default"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head
          element"
    data-cy="head"
  >
    <div
      className="content "
      data-cy="circle-small"
      data-test-state="default"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head
          element"
        data-cy="head-small"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-cy="letter-small"
        >
          h
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-cy="index"
      />
      <div
        className="text text_type_input text_color_input mt-4
          absolute
          tail30
          element
        "
        data-cy="tail-small"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    >
      H
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4
          absolute
          tail30
          element
        "
    data-cy="tail"
  />
</div>
`;

exports[`Кружок с react-элементом в tail 1`] = `
<div
  className="content "
  data-cy="circle"
  data-test-state="default"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head
          element"
    data-cy="head"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    >
      T
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4
          absolute
          tail30
          element
        "
    data-cy="tail"
  >
    <div
      className="content "
      data-cy="circle"
      data-test-state="default"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head
          element"
        data-cy="head"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-cy="letter"
        >
          t
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-cy="index"
      />
      <div
        className="text text_type_input text_color_input mt-4
          absolute
          tail30
          element
        "
        data-cy="tail"
      />
    </div>
  </div>
</div>
`;

exports[`Кружок с tail 1`] = `
<div
  className="content "
  data-cy="circle"
  data-test-state="default"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head
          element"
    data-cy="head"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    >
      T
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4
          absolute
          tail30
          string
        "
    data-cy="tail"
  >
    t
  </div>
</div>
`;

exports[`Кружок с буквами 1`] = `
<div
  className="content "
  data-cy="circle"
  data-test-state="default"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head
          element"
    data-cy="head"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    >
      123
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4
          absolute
          tail30
          element
        "
    data-cy="tail"
  />
</div>
`;

exports[`Кружок с пропcом isSmall ===  true 1`] = `
<div
  className="content "
  data-cy="circle-small"
  data-test-state="default"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head
          element"
    data-cy="head-small"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter-small"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4
          absolute
          tail30
          element
        "
    data-cy="tail-small"
  />
</div>
`;
